section2/set9/exercises/04-verbose-mode/main.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2018 Inanc Gumus
+// Learn Go Programming Course
+// License: https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// For more tutorials  : https://learngoprogramming.com
+// In-person training  : https://www.linkedin.com/in/inancgumus/
+// Follow me on twitter: https://twitter.com/inancgumus
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"time"
+)
+
+// ---------------------------------------------------------
+// EXERCISE: Verbose Mode
+//
+//  When the player runs the game like this:
+//
+//    go run main.go -v 4
+//
+//  Display each generated random number:
+
+//    1 3 4 🎉  YOU WIN!
+//
+//  In this example, computer picks 1, 3, and 4. And the
+//  player wins.
+//
+// HINT
+//  You need to get and interpret the command-line arguments.
+// ---------------------------------------------------------
+
+func main() {
+	const (
+		maxTurns = 5
+		usage    = `Welcome to the Lucky Number Game!
+
+The program will pick %d random numbers.
+Your mission is to guess one of those numbers.
+
+The greater your number is, the harder it gets.
+
+Want to play?`
+	)
+
+	rand.Seed(time.Now().UnixNano())
+
+	args := os.Args[1:]
+	var guess int
+	var err error
```

Comment:
declare variables just before where they are used; here err is used later on like 63

Reply:
guess and err need to be in the global scope, and so they should be accessible by rest of the program

Reply:
unless you are learning global variables, you should define as close to the place you use them